chroma-manager/tests/integration/shared_storage_configuration/test_filesystem_same_name.py

```diff
@@ -73,18 +72,47 @@ def create_for_mgs(name, hosts_to_use, reformat=False):
         # Now remove any zfs datasets, this is a topic to be discussed, but until we remove the datasets
         # we cannot create a new filesystem. If IML does it directly as part of remove filesystem which it could
         # then removing the filesystem would be truly unrecoverable and people might not like that.
-        datasets = [ost['volume']['volume_nodes'][0]['path'] for ost in fs['osts']]
-        datasets.extend([mdt['volume']['volume_nodes'][0]['path'] for mdt in fs['mdts']])
+        datasets = [
+            ost['volume']['volume_nodes'][0]['path'] for ost in fs['osts']
+        ]
+        datasets.extend(
+            [mdt['volume']['volume_nodes'][0]['path'] for mdt in fs['mdts']])
 
         # Filter out the paths by removing anything with a leading /.
-        datasets = [dataset for dataset in datasets if dataset.startswith('/') is False]
-
-        self.remote_operations.stop_agents(s['address'] for s in self.TEST_SERVERS[:4])
-        self.cleanup_zfs_pools(self.TEST_SERVERS[:4],
-                               self.CZP_REMOVEDATASETS | self.CZP_EXPORTPOOLS,
-                               datasets,
-                               True)
-        self.remote_operations.start_agents(s['address'] for s in self.TEST_SERVERS[:4])
+        datasets = [
+            dataset for dataset in datasets if dataset.startswith('/') is False
+        ]
+        datasets.sort(key=lambda x: -len(x))
+
+        pools = map(lambda x: x.split('/')[0], datasets)
+
+        fqdns = [x['fqdn'] for x in self.TEST_SERVERS[:4]]
+
+        self.remote_operations.stop_agents(s['address']
+                                           for s in self.TEST_SERVERS[:4])
+        for pool in pools:
+            self.execute_simultaneous_commands(
+                ['zpool import %s' % pool],
+                fqdns,
+                'import pool %s' % pool,
+                expected_return_code=None)
+
+        for zpool_dataset in datasets:
+            self.execute_simultaneous_commands(
+                ['zfs destroy %s' % zpool_dataset],
+                fqdns,
+                'destroy zfs dataset %s' % zpool_dataset,
+                expected_return_code=None)
+
+        for pool in pools:
+            self.execute_simultaneous_commands(
+                ['zpool export %s' % pool],
+                fqdns,
+                'export pool %s' % pool,
+                expected_return_code=None)
+
+        self.remote_operations.start_agents(s['address']
+                                            for s in self.TEST_SERVERS[:4])
```

Comment:
A ticket for this assumption that there are only 4 lustre servers would be good too i think.  Just to keep track of the technical debt.

Reply:
Will do.

Reply:
Opened: https://github.com/intel-hpdd/intel-manager-for-lustre/issues/352 as a general issue on SSI / EFS cleanup (feel free to add to it).